app/data_model/progress_store.py

```diff
@@ -64,14 +64,30 @@ def is_dirty(self) -> bool:
     def is_section_complete(
         self, section_id: str, list_item_id: Optional[str] = None
     ) -> bool:
-        return (section_id, list_item_id) in self._completed_section_keys()
+        return (section_id, list_item_id) in self.section_keys_by_status(
+            {CompletionStatus.COMPLETED}
+        )
 
-    def _completed_section_keys(self) -> List[Tuple[str, Optional[str]]]:
-        return [
+    def section_keys_by_status(
+        self, statuses: Iterable[str]
+    ) -> Set[Tuple[str, Optional[str]]]:
+        return {
             section_key
             for section_key, section_progress in self._progress.items()
-            if section_progress.status == CompletionStatus.COMPLETED
-        ]
+            if section_progress.status in statuses
+        }
+
+    def get_in_progress_and_completed_sections(
+        self, filter_by: Set[str] = None
```

Comment:
I can't see where this method is called with `filter_by` apart from a test?.

Reply:
Yeah good shout - this was evaluating every completed or in progress section. I've updated in [da77aab5](https://github.com/ONSdigital/eq-questionnaire-runner/pull/9/commits/da7aab520465aca3e3d28bcf5ccfd6ccb20f4a7d)